contrib/internal/telemetrytest/telemetry_test.go

```diff
@@ -0,0 +1,108 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+package telemetrytest
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/contrib/gorilla/mux"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/telemetry"
+)
+
+// TestIntegrationInfo verifies that an integration leveraging instrumentation telemetry
+// sends the correct data to the telemetry client.
+func TestIntegrationInfo(t *testing.T) {
+	mux.NewRouter()
+	integrations := telemetry.Integrations()
+	assert.Len(t, integrations, 1)
+	assert.Equal(t, integrations[0].Name, "gorilla/mux")
+	assert.True(t, integrations[0].Enabled)
+}
+
+type contribPkg struct {
+	ImportPath string
+	Name       string
+	Imports    []string
+}
+
+var TelemetryImport = "gopkg.in/DataDog/dd-trace-go.v1/internal/telemetry"
+
+func (p *contribPkg) hasTelemetryImport() bool {
+	for _, imp := range p.Imports {
+		if imp == TelemetryImport {
+			return true
+		}
+	}
+	return false
+}
+
+// parseContribPath takes the current path and returns the relative path
+// to the contrib folder. The current path must be a sub-directory of the
+// contrib folder or the parent dd-trace-go directory.
+func parseContribPath(path string) (string, error) {
+	if filepath.Base(path) == "dd-trace-go" {
+		return "./contrib", nil
+	}
+	dirs := strings.Split(path, "/")
+	for i, dir := range dirs {
+		if dir == "contrib" {
+			contribPath := filepath.Join(dirs[:i+1]...)
+			rel, err := filepath.Rel(strings.TrimPrefix(path, "/"), contribPath)
+			if err != nil {
+				return "", err
+			}
+			return rel, nil
+		}
+	}
+	return "", fmt.Errorf("contrib was not found as a parent folder and current working directory is not dd-trace-go")
+}
+
+// TestTelemetryEnabled verifies that the expected contrib packages leverage instrumentation telemetry
+func TestTelemetryEnabled(t *testing.T) {
+	tracked := map[string]struct{}{"mux": {}}
```

Comment:
so presumably, this will be removed when the `init()` functions are added to all of the other contrib libraries, right?

Reply:
Yep!